File: packages/flutter/lib/core/l10n/language_id.dart
import 'package:allowance_questboard/core/messages/locale_string.dart';
import 'package:allowance_questboard/core/state/base_id.dart' show BaseId;

/// 言語IDクラス
class LanguageId extends BaseId {
  LanguageId(super.value);

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is LanguageId &&
          runtimeType == other.runtimeType &&
          value == other.value;

  @override
  int get hashCode => value.hashCode;

  @override
  String toString() => 'LanguageId($value)';

  @override
  LocaleString get valueName => LocaleString(
    en: 'Language ID',
    ja: '言語ID',
  );
}

File: packages/api/aqapi/language/domain/value_object/language_id.py
from dataclasses import dataclass
from typing import TYPE_CHECKING, override

from aqapi.core.domain.value_object.base_id import BaseId
from aqapi.core.messages.locale_string import LocaleString


class LanguageId(BaseId):
    """言語IDの値オブジェクト"""
    def __init__(self, value: int):
        super().__init__(value)

    @property
    @override
    def _value_name(self) -> LocaleString:
        return LocaleString(
            ja="言語ID",
            en="Language ID"
        )
